deicode/rpca.py

```diff
@@ -2,17 +2,19 @@
 import skbio
 import numpy as np
 import pandas as pd
+from typing import Union
 from deicode.matrix_completion import MatrixCompletion
 from deicode.preprocessing import rclr
 from deicode._rpca_defaults import (DEFAULT_RANK, DEFAULT_MSC, DEFAULT_MFC,
-                                    DEFAULT_ITERATIONS)
+                                    DEFAULT_ITERATIONS, DEFAULT_MFF)
 from scipy.linalg import svd
 
 
 def rpca(table: biom.Table,
-         n_components: int = DEFAULT_RANK,
+         n_components: Union[int, str] = DEFAULT_RANK,
```

Comment:
is this complex type casting still necessary?

Reply:
The way I have it set up, to prevent too much copy-pasta, is that `auto_rpca` calls `rpca` but with n_components flagged as `"auto"`. However, the types are restricted in the CLI so you couldn't run `rpca` with the input `"auto"` in QIIME or in the standalone. So technically that `rpca` function still accepts both types.